### PR TITLE
Add log wrapper

### PR DESCRIPTION
**Pull request by nikitin-da**
Part of #79 
Just only for `StorIODb`.
@artem-zinnatullin , did you meant something like this?

### CODE REVIEW

File: storio/src/main/java/com/pushtorefresh/storio/ILogger.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.pushtorefresh.storio;
+
+public interface ILogger {
```
Comment (artem-zinnatullin) [inline storio/src/main/java/com/pushtorefresh/storio/ILogger.java]: It's not a `C#`, I think we can call it just `Logger` :smile:

File: storio/src/main/java/com/pushtorefresh/storio/ILogger.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    void v(String tag, String message);
```
Comment (artem-zinnatullin) [inline storio/src/main/java/com/pushtorefresh/storio/ILogger.java]: Also, I think we can left only messages of different levels, without tags, tag can be set as final property in Logger impl constructor, it's not logger for application, just small library :)

File: storio/src/main/java/com/pushtorefresh/storio/ILogger.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    void i(String tag, String message);
+
+    void w(String tag, String message);
+
+    void e(String tag, String message);
+
+    void e(String tag, String message, Throwable tr);
+
+    int wtf(String tag, String message);
```
Comment (artem-zinnatullin) [inline storio/src/main/java/com/pushtorefresh/storio/ILogger.java]: Why return type `int`? How it used by `android.util.Log`?

File: storio/src/main/java/com/pushtorefresh/storio/db/impl/StorIOSQLiteDbImpl.java [hunk 4]
```diff
@@ -155,6 +155,8 @@ public void notifyAboutChanges(@NonNull Changes changes) {
             // Notifying about changes requires RxJava, if RxJava is not available -> skip notification
             if (changesBus != null) {
                 changesBus.onNext(changes);
+            } else {
+                loggi.d("Changes was occurred, but changesBus is null. " + changes);
```
Comment (artem-zinnatullin) [inline storio/src/main/java/com/pushtorefresh/storio/db/impl/StorIOSQLiteDbImpl.java]: This log will be shown pretty often..

File: storio/src/main/java/com/pushtorefresh/storio/ILogger.java (hunk 1 above)
Comment (artem-zinnatullin) [inline storio/src/main/java/com/pushtorefresh/storio/ILogger.java]: `@NonNull/@Nullable` annotations for methods params

File: storio/src/main/java/com/pushtorefresh/storio/Loggi.java [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+import android.text.TextUtils;
+import android.util.Log;
+
+/**
+ * Simple android Log proxy, includes default TAG and provides ability to turn logs on/off
+ * <p/>
+ * Call it "Loggi, come here!", you can think that Loggi is your little dog :)
+ */
+@SuppressWarnings({ "PMD.ProtectLogD", "PMD.ProtectLogV" })
+public class Loggi {
```
Comment (artem-zinnatullin) [inline storio/src/main/java/com/pushtorefresh/storio/Loggi.java]: Hm, I know this class..

Comment (artem-zinnatullin) [inline storio/src/main/java/com/pushtorefresh/storio/Loggi.java]: `@NonNull/@Nullable` for method params?)

File: storio/src/main/java/com/pushtorefresh/storio/ILogger.java (hunk 3 above)
Comment (nikitin-da) [inline storio/src/main/java/com/pushtorefresh/storio/ILogger.java]: It's the number of bytes written, from the underlying println.
I will remove.

File: storio/src/main/java/com/pushtorefresh/storio/Loggi.java (hunk 5 above)
Comment (nikitin-da) [inline storio/src/main/java/com/pushtorefresh/storio/Loggi.java]: There is no any `author` tags in storIO's javadocs, but it's `artem_zin`'s creation. 
Have you ever heard about him?

File: storio/src/main/java/com/pushtorefresh/storio/ILogger.java (hunk 3 above)
Comment (artem-zinnatullin) [inline storio/src/main/java/com/pushtorefresh/storio/ILogger.java]: :+1:

File: storio/src/main/java/com/pushtorefresh/storio/Loggi.java (hunk 5 above)
Comment (artem-zinnatullin) [inline storio/src/main/java/com/pushtorefresh/storio/Loggi.java]: Nope, I should find him

**Comment by nikitin-da** [comment]
Thank you for advises! 
But how we can avoid public getter for internal loggi?

**Comment by artem-zinnatullin** [comment]
Just hide it into Internal class and write Java doc where you can say that
Loggi is for internal usage

вс, 22 марта 2015, 17:34, Dmitrii Nikitin <notifications@github.com>:

**Comment by nikitin-da** [comment]
What is better way to get ride troubles with tests.
Now I need mock Internal.getLoggi() method to avoid NPE and Loggi.isEnabled()

**Comment by artem-zinnatullin** [comment]
I don't know man, sorry :(

пн, 23 марта 2015 г. в 0:38, Dmitrii Nikitin <notifications@github.com>:

**Comment by artem-zinnatullin** [comment]
But I'd like to see your solutions :)

пн, 23 марта 2015 г. в 1:27, Artem Zinnatullin <artem.zinnatullin@gmail.com

**Comment by artem-zinnatullin** [comment]
retest this please

**Comment by artem-zinnatullin** [comment]
@nikitin-da can you please merge `master` into this branch, it does not know about `ci/ci.sh`

**Comment by nikitin-da** [comment]
yep

**Comment by artem-zinnatullin** [comment]
@nikitin-da oh, wait

**Comment by artem-zinnatullin** [comment]
@nikitin-da Jenkins automatically merges it before building! There is problem with tests

**Comment by public-ci-pushtorefresh** [comment]
Can one of the admins verify this patch?

**Comment by artem-zinnatullin** [comment]
retest this please

**Comment by artem-zinnatullin** [comment]
retest this please

**Comment by nikitin-da** [comment]
what?

**Comment by nikitin-da** [comment]
it speaks!!!

**Comment by artem-zinnatullin** [comment]
retest this please

**Comment by nikitin-da** [comment]
I have hidden loggi from client. 
@artem-zinnatullin PTAL

**Comment by artem-zinnatullin** [comment]
Okay, great! But I think we'll refactor this a little bit before release :)

:+1:

**Pull request merged**